src/App.tsx

```diff
@@ -1,222 +1,124 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import { getTodos, removeTodo } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
 
+import { ErrorMessage } from './types/ErrorMessage';
+import { Status } from './types/Status';
+import { Todo } from './types/Todo';
+
+import { AddField } from './components/AddFiled/AddField';
+import { Filter } from './components/Filter/Filter';
+import { Errors } from './components/Errors/Errors';
+import { Todos } from './components/Todos/Todos';
+
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [isTodosAvailable, setIsTodosAvailable] = useState(false);
```

Comment:
maybe it would be easier to check for the length of the todo's array instead this variable